source/velocity_boundary_conditions/interface.cc

```diff
@@ -70,7 +70,8 @@ namespace aspect
     }
 
 
-
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wdeprecated-declarations"
```

Comment:
Oh, good point. Totally did not pay attention to this kind of problem :-(

What problem does this particular part of the patch address?